Azure Pipelines Windows Driver Build Fails

### Describe the bug

AZP recently updated to the newest WDK it seems, and it is now failing to build because of the following:
```
ClCompile:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.26.28801\bin\HostX86\x64\CL.exe /c /I..\inc /Id:\a\1\s\build\winkernel\x64_Debug_schannel\inc /Id:\a\1\s\build\winkernel\x64_Debug_schannel\obj\platform.kernel\ /Id:\a\1\s\build\winkernel\x64_Debug_schannel\obj\platform.kernel\ /Zi /nologo /W4 /WX /diagnostics:column /MP /Od /Oi /Ot /Oy- /GL /D CODE_ANALYSIS /D QUIC_EVENTS_MANIFEST_ETW /D QUIC_LOGS_MANIFEST_ETW /D SECURITY_KERNEL /D SECURITY_WIN32 /D _DEBUG /D _WIN64 /D _AMD64_ /D AMD64 /D DEPRECATE_DDK_FUNCTIONS=1 /D MSC_NOOPT /D _WIN32_WINNT=0x0A00 /D WINVER=0x0A00 /D WINNT=1 /D NTDDI_VERSION=0x0A000008 /D DBG=1 /D KMDF_VERSION_MAJOR=1 /D KMDF_VERSION_MINOR=15 /GF /Gm- /MDd /Zp8 /GS /guard:cf /Gy /fp:precise /Qspectre /Zc:wchar_t- /Zc:forScope /Zc:inline /GR- /Fo"d:\a\1\s\build\winkernel\x64_Debug_schannel\obj\platform.kernel\\" /Fd"d:\a\1\s\build\winkernel\x64_Debug_schannel\obj\platform.kernel\platform.kernel.pdb" /Gz /wd4748 /wd4603 /wd4627 /wd4986 /wd4987 /FI"C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\shared\warning.h" /analyze /analyze:projectdirectory"d:\a\1\s\src\platform\\" /analyze:rulesetdirectory";C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Team Tools\Static Analysis Tools\\Rule Sets;" /analyze:ruleset "C:\Program Files (x86)\Windows Kits\10\CodeAnalysis\DriverMinimumRules.ruleset" /analyze:quiet /analyze:stacksize1024 /analyze:plugin"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.26.28801\bin\HostX86\x86\EspXEngine.dll" /analyze:plugin"C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x86\WindowsPrefast.dll" /analyze:plugin"C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x86\drivers.dll" /FC /errorReport:queue /Gw /kernel /ZH:SHA_256 -d2jumptablerdata -d2epilogunwindrequirev2 datapath_winkernel.c hashtable.c platform_winkernel.c storage_winkernel.c tls_schannel.c toeplitz.c
  datapath_winkernel.c
  hashtable.c
  platform_winkernel.c
  storage_winkernel.c
##[error]src\platform\storage_winkernel.c(220,5): Error C2220: the following warning is treated as an error
d:\a\1\s\src\platform\storage_winkernel.c(220,5): error C2220: the following warning is treated as an error [d:\a\1\s\src\platform\platform.kernel.vcxproj]
##[warning]src\platform\storage_winkernel.c(220,5): Warning C4996: 'ExInitializeWorkItem': was declared deprecated
d:\a\1\s\src\platform\storage_winkernel.c(220,5): warning C4996: 'ExInitializeWorkItem': was declared deprecated [d:\a\1\s\src\platform\platform.kernel.vcxproj]
```
We need to ignore the deprecated warning/error for now.

--- a/src/platform/storage_winkernel.c
+++ b/src/platform/storage_winkernel.c
@@ -216,11 +216,14 @@ QuicStorageOpen(
     QuicLockInitialize(&Storage->Lock);
     Storage->Callback = Callback;
     Storage->CallbackContext = CallbackContext;
-    
+
+#pragma warning(push)
+#pragma warning(disable: 4996)
     ExInitializeWorkItem(
         &Storage->WorkItem,
         QuicStorageRegKeyChangeCallback,
         Storage);
+#pragma warning(pop)
 
     Status =
         ZwOpenKey(

